marshal.go

```diff
@@ -44,6 +44,51 @@ type Unmarshaler interface {
 
 // Marshal returns the CQL encoding of the value for the Cassandra
 // internal type described by the info parameter.
+//
+// nil is serialized as CQL null.
+// If value implements Marshaler, its MarshalCQL method is called to marshal the data.
+// If value is a pointer, the pointed-to value is marshaled.
+//
+// Supported conversions are as follows:
+//
+//  CQL type                    | Go type (value)    | Note
+//  varchar, ascii, blob, text  | string, []byte     |
+//  boolean                     | bool               |
+//  tinyint, smallint, int      | integer types      |
+//  tinyint, smallint, int      | string             | formatted as base 10 number
+//  bigint, counter             | integer types      |
+//  bigint, counter             | big.Int            |
+//  bigint, counter             | string             | formatted as base 10 number
+//  float                       | float32            |
+//  double                      | float64            |
+//  decimal                     | inf.Dec            |
+//  time                        | int64              | nanoseconds since start of day
+//  time                        | time.Duration      | duration since start of day
+//  timestamp                   | int64              | milliseconds since Unix epoch
+//  timestamp                   | time.Time          |
+//  list, set                   | slice, array       |
+//  list, set                   | map[X]struct{}     |
+//  map                         | map[X]Y            |
+//  uuid, timeuuid              | UUID               |
+//  uuid, timeuuid              | []byte             | raw UUID bytes, length must be 16 bytes
+//  uuid, timeuuid              | string             | hex representation, see ParseUUID
+//  varint                      | integer types      |
+//  varint                      | big.Int            |
+//  varint                      | string             | value of number in decimal notation
+//  inet                        | net.IP             |
+//  inet                        | string             | IPv4 or IPv6 address string
+//  tuple                       | slice, array       |
+//  tuple                       | struct             | fields are marshaled in order of declaration
+//  user-defined type           | UDTMarshaler       | MarshalUDT is called
+//  user-defined type           | map[string]interface{} |
+//  user-defined type           | struct             | cql tags are used for field names
```

Comment:
**though**: this might be more accurate:
```suggestion
//  user-defined type           | struct             | struct fields cql tags are used for column names
```

Reply:
Updated, thanks!